mongomock/collection.py

```diff
@@ -1649,6 +1649,18 @@ def _extend_collection(out_collection, field, expression):
                 if k == '$match':
                     out_collection = [doc for doc in out_collection
                                       if filter_applies(v, doc)]
+                elif k == '$lookup':
+                    foreign_name = stage['$lookup']['from']
+                    local_field = stage['$lookup']['localField']
+                    foreign_field = stage['$lookup']['foreignField']
+                    local_name = stage['$lookup']['as']
+                    foreign_collection = self.database.get_collection(foreign_name)
+                    for doc in out_collection:
+                        query = doc[local_field]
+                        if type(doc[local_field]) is list:
+                            query = {'$in': query}
+                        matches = foreign_collection.find({foreign_field: query})
```

Comment:
I'm not sure how `$lookup` should work, please add a link to the MongoDB page describing the feature to your PR. Also if some docs have some weird values in the local_field like `{'$exists': True}`.

Reply:
I added the link with restriction in my original comment, thank you for the suggestion.
As for the weird values (specifically field names prefixed with `$`), see https://docs.mongodb.com/manual/reference/limits/#Restrictions-on-Field-Names